### PR TITLE
Implement WordPress Object Cache storage

### PR DESCRIPTION
**Pull request by colinodell**
Implements cache storage using WP's Object Cache - useful when Guzzle is used on a WordPress site.

### CODE REVIEW

File: src/Storage/WordPressObjectCacheStorage.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+     * @var string
+     */
+    private $group;
+
+    /**
+     * @param string $group
+     */
+    public function __construct($group = 'guzzle')
+    {
+        $this->group = $group;
```
Comment (colinodell) [inline src/Storage/WordPressObjectCacheStorage.php]: WordPress essentially namespaces keys into different groups to help prevent conflicts.  I think a default group of `guzzle` makes sense, but we should allow users to change that if needed.

File: src/Storage/WordPressObjectCacheStorage.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    /**
+     * @param string $key
+     * @param CacheEntry $data
+     *
+     * @return bool
+     */
+    public function save($key, CacheEntry $data)
+    {
+        try {
+            return wp_cache_set($key, serialize($data), $this->group, $data->getTTL());
```
Comment (colinodell) [inline src/Storage/WordPressObjectCacheStorage.php]: https://codex.wordpress.org/Function_Reference/wp_cache_set

File: src/Storage/WordPressObjectCacheStorage.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    /**
+     * @param string $key
+     *
+     * @return CacheEntry|null the data or false
+     */
+    public function fetch($key)
+    {
+        try {
+            $cache = unserialize(wp_cache_get($key, $this->group));
```
Comment (colinodell) [inline src/Storage/WordPressObjectCacheStorage.php]: https://codex.wordpress.org/Function_Reference/wp_cache_get

**Comment by Kevinrob** [comment]
Thanks for this PR! And for the comments, it will help me to review it.  
I will check it asap.

**Pull request merged**